libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c

```diff
@@ -940,18 +1092,23 @@ void test_pkcs11_C_CreateObjectECPrivKey( void )
                               sizeof( xPrivateKeyTemplate ) / sizeof( CK_ATTRIBUTE ),
                               &xObject );
 
-    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+    TEST_ASSERT_EQUAL( CKR_FUNCTION_FAILED, xResult );
 
-    xResult = prvCloseSession( &xSession );
-    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+    /* Malloc fails in prvSaveDerKeyToPal */
+    pvPortMalloc_Stub( pvPkcs11MallocCb );
+    pvPortMalloc_IgnoreAndReturn( NULL );
+    mbedtls_pk_write_key_der_IgnoreAndReturn( 1 );
+    xResult = C_CreateObject( xSession,
+                              ( CK_ATTRIBUTE_PTR ) &xPrivateKeyTemplate,
+                              sizeof( xPrivateKeyTemplate ) / sizeof( CK_ATTRIBUTE ),
+                              &xObject );
 
-    xResult = prvUninitializePkcs11();
-    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+    TEST_ASSERT_EQUAL( CKR_HOST_MEMORY, xResult );
 
-    /* TODO: Remove this. Currently memory is malloc'd by PKCS #11 and free'd in the
-     * MbedTLS stack, but those functions are stubbed out. Need to update this test case
-     * to have a memory life cycle that is properly mocked. */
-    usMallocFreeCalls--;
+    prvCommonDeinitStubs();
+
+    /* Since malloc does not require a NULL check, the stub was called one time too many. */
+    usMallocFreeCalls++;
```

Comment:
Please avoid this by letting the malloc stub check for null and not count when it is null.